### PR TITLE
20161019-123400-swap-connections-and-persist-commands

### PR DESCRIPTION
**Pull request by andySigler**
This PR:

1. Adds argument to `robot.run()`, where `robot.run(mode="live")` runs on serial connection, and `robot.run()` or `robot.run(mode="simulate")` runs on virtual smoothie
2. Commands are no longer popped but are persisted between calls to `robot.run()`, so that same protocol can be run multiple times
3. After `robot.run()` complete, the device connection is automatically switched back to the real ("live") smoothie board
4. To erase all Commands, call `robot.clear()`, or `robot.reset()` to also clear the deck and instruments

### CODE REVIEW

File: opentrons_sdk/robot/robot.py [hunk 1]
```diff
@@ -53,9 +72,12 @@ def add_instrument(self, axis, instrument):
         axis = axis.upper()
         self._instruments[axis] = instrument
 
-    def add_warning(self, warning_msg: str):
+    def add_warning(self, warning_msg):
         self._runtime_warnings.append(warning_msg)
 
+    def get_warnings(self):
+        return list(self._runtime_warnings)
```
Comment (SimplyAhmazing) [inline opentrons_sdk/robot/robot.py]: `_runtimewarnings` is already a list right?

File: opentrons_sdk/robot/robot.py [hunk 2]
```diff
@@ -100,16 +122,55 @@ def flip_coordinates(self, coordinates):
...
+
+        return None
+
+    def get_virtual_device(self, port=None, options=None):
+        default_options = {
+            'limit_switches': True,
+            'firmware': 'v1.0.5',
+            'config': {
+                'ot_version': 'one_pro',
+                'version': 'v1.0.3',        # config version
```
Comment (SimplyAhmazing) [inline opentrons_sdk/robot/robot.py]: kind of thinking we should rename this key to `config_file_version`

File: opentrons_sdk/robot/robot.py (hunk 1 above)
Comment (andySigler) [inline opentrons_sdk/robot/robot.py]: the thought was to copy it incase the receiver wants to for example `.pop()` from it

File: opentrons_sdk/robot/robot.py (hunk 2 above)
Comment (andySigler) [inline opentrons_sdk/robot/robot.py]: the config dict there are the literal key-value pairs inside the virtual config file. The `version` key is what it's called in the real config file

File: tests/opentrons_sdk/labware/test_pipette.py [hunk 3]
```diff
@@ -251,7 +252,7 @@ def test_non_empty_aspirate(self):
 
     def test_aspirate_no_args(self):
         self.p200.aspirate()
-        self.robot.run()
+        self.robot.run(mode='live')
```
Comment (SimplyAhmazing) [inline tests/opentrons_sdk/labware/test_pipette.py]: hmm.. shouldn't all tests run on `simulate`?

**Comment by SimplyAhmazing** [comment]
Besides comments, LGTM

File: tests/opentrons_sdk/labware/test_pipette.py (hunk 3 above)
Comment (andySigler) [inline tests/opentrons_sdk/labware/test_pipette.py]: the tests are using non-simulation features, like jogging and calibrating. To get around this, we set the "live" port to be the `VirtualSmoothie`

**Pull request merged**